js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/flattenEvaluations.tsx

```diff
@@ -1,42 +1,64 @@
 import {ConditionType} from './PolicyEvaluationCondition';
-import {AssetConditionEvaluation} from './types';
+import {
+  AssetConditionEvaluationRecordFragment,
+  PartitionedAssetConditionEvaluationNodeFragment,
+  SpecificPartitionAssetConditionEvaluationNodeFragment,
+  UnpartitionedAssetConditionEvaluationNodeFragment,
+} from './types/GetEvaluationsQuery.types';
 
-type FlattenedConditionEvaluation<T> = {
+export type FlattenedConditionEvaluation<T> = {
   evaluation: T;
   id: number;
   parentId: number | null;
   depth: number;
   type: ConditionType;
 };
 
-export const flattenEvaluations = <T extends AssetConditionEvaluation>(rootEvaluation: T) => {
-  const all: FlattenedConditionEvaluation<T>[] = [];
+type Evaluation =
+  | PartitionedAssetConditionEvaluationNodeFragment
+  | UnpartitionedAssetConditionEvaluationNodeFragment
+  | SpecificPartitionAssetConditionEvaluationNodeFragment;
+
+type FlattenedEvaluation =
+  | FlattenedConditionEvaluation<PartitionedAssetConditionEvaluationNodeFragment>
+  | FlattenedConditionEvaluation<UnpartitionedAssetConditionEvaluationNodeFragment>
+  | FlattenedConditionEvaluation<SpecificPartitionAssetConditionEvaluationNodeFragment>;
+
+export const flattenEvaluations = (
+  evaluationRecord: Pick<AssetConditionEvaluationRecordFragment, 'evaluation'>,
+) => {
```

Comment:
Why the change away from the generic type?

Reply:
It doesn't work unfortunately because of how the backend is typed